internal/server/mcp.go

```diff
@@ -71,6 +84,30 @@ func (m *sseManager) remove(id string) {
 	m.mu.Unlock()
 }
 
+func (m *sseManager) cleanupRoutine(ctx context.Context) {
+	timeout := 10 * time.Minute
+	ticker := time.NewTicker(timeout)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		m.mu.Lock()
+		now := time.Now()
+		for id, sess := range m.sseSessions {
+			if now.Sub(sess.lastActive) > timeout {
+				delete(m.sseSessions, id)
+			}
+		}
+		m.mu.Unlock()
+	}
+
+	// to shutdown goroutine if ctx is cancelled
+	select {
+	case <-ctx.Done():
+		return
+	default:
+	}
```

Comment:
I don't think this is right. you will only get here once the `range ticker.C` is closed. 

Shouldn't you listen for this in between ticks?